### PR TITLE
Analysis/ steps error

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Analysis error is marked on the correct step rather than defaulting to the "Queued" step

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java [hunk 1]
```diff
@@ -3,7 +3,11 @@
 import java.security.Principal;
 import java.util.*;
```
Comment (joshsadam) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java]: There is a `PREVIEWS` map at the top that is never used, can you remove it here please.

Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java]: I'll remove it in the cleanup branch that I have going

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java [hunk 2]
```diff
@@ -158,9 +168,47 @@ public String getDetailsPage(@PathVariable Long submissionId, Model model, final
 		model.addAttribute("analysisType", analysisType);
 		model.addAttribute("mailConfigured", emailController.isMailConfigured());
 
+		if(submission.getAnalysisState()==AnalysisState.ERROR) {
```
Comment (joshsadam) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java]: Your formatting is off for this (no spaces around the `==`), can you fix it please.

Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java]: Fixed in 8448f0a75c32d57c156973710e4346c707a8567d

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java [hunk 3]
```diff
@@ -158,9 +168,47 @@ public String getDetailsPage(@PathVariable Long submissionId, Model model, final
...
+	 * @param submissionId {@link Long} identifier for an {@link AnalysisSubmission}
+	 * @return {@link String} State of analysis prior to error
+	 */
+	private AnalysisState getPreviousStateBeforeError(Long submissionId) {
+		AuditReader audit =AuditReaderFactory.get(entityManagerFactory.createEntityManager());
+		AnalysisSubmission previousRevision = null;
+
+		if(audit != null) {
+			// Get revisions from the analysis submission audit table for the submission
+			List auditResultSet = audit.createQuery()
```
Comment (joshsadam) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java]: Is it possible to parameterize this list?

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java [hunk 4]
```diff
@@ -158,9 +168,47 @@ public String getDetailsPage(@PathVariable Long submissionId, Model model, final
...
+			// prior is set to the previousRevision and it's state is set in the model
+			for (Object auditResult : auditResultSet) {
+				AnalysisSubmission auditedSubmission = (AnalysisSubmission) auditResult;
+				if (auditedSubmission.getAnalysisState() == AnalysisState.ERROR) {
+					break;
+				}
+				previousRevision = auditedSubmission;
+			}
+		}
+		if(previousRevision==null) {
```
Comment (joshsadam) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java]: Formatting, why do you hate spaces so much?

Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java]: Fixed in the previous commit. Forgot to run the intellij formatter on the method :)

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java (hunk 3 above)
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java]: Added in 47d5d9d0375e49f3d1f50ccc63a12242db6c13e2

Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java]: Nice, thanks

**Pull request merged**